setup.py

```diff
@@ -196,11 +196,11 @@ def run(self):
 
         # Build a static binary on Windows to avoid missing DLLs from MinGW,
         # i.e., libgcc_s_seh-1.dll, libstdc++-6.dll, etc.
-        BUILDING_FOR_WINDOWS = (
+        BUILD_STATIC = (
             os.environ.get("GOOS") == "windows" or sys.platform == "win32"
```

Comment:
This condition can now be modified to check if we are building for MUSL. There is just one way to do that, which is to check if the `CC` and `CXX` environment variable strings contain `linux-musl`, e.g., if CC is set to "zig cc -target x86_64-linux-musl", and so on.

Reply:
@agriyakhetarpal I've updated the condition, does it need to check if CC is set to "zig cc -target x86_64-linux-musl" exactly?

Reply:
It could be `aarch64-linux-musl` as well, and not just `x86_64`, so we shouldn't need to check for an exact match. This commit should be okay.